fem/fespace.hpp

```diff
@@ -724,6 +865,20 @@ class FiniteElementSpace
       return dynamic_cast<const L2_FECollection*>(fec) != NULL;
    }
 
+   /** In variable order spaces on nonconforming meshes, this function controls
+       whether strict conformity is enforced in cases where master edges/faces
+       have higher polynomial order than their refined slaves. In the default
+       (strict) case, the master polynomial order is reduced to that of the
+       lowest order slave, so the slave can interpolate the master side exactly.
+       If relaxed == true, discontinuities in the solution in such cases are
+       allowed and the master side is not restricted. */
+   void SetRelaxedHpConformity(bool relaxed = true)
```

Comment:
This is a function that general users need to understand on a high level, even if they do not understand anything about the implementation. The case described here may not be entirely clear to users. It may help to explain that "master edges/faces" are coarser than their slaves in h-refinement (if I understand correctly). The reader might confuse h- and p-refinement, so "refined slaves" might allow them to think of "p-refined slaves" with higher order. This could be clarified by explicitly saying "h-refined slaves." It could also be mentioned that the case of equal or lower order on the master always has strict conformity. Also, it is not clear why `relaxed == false` would ever be desirable, and what it means exactly. It is not the same as complete discontinuity (as in L2 spaces), as only a subset of DOFs are allowed to be discontinuous. Is there a use case for `false` for which this is designed?

Reply:
I have updated this comment to avoid the internal terminology, added explicit mentions that these are coarse/fine NC neighbors (not p-refinements), and added a link to a github comment with pictures.

The relaxed case was initially considered for special uses only, but it turns out it will be commonly needed, as seen e.g. in the hptest miniapp, where the strict conformity is overconstraining the solution in some cases (see discussions on the L-shaped problem in the `p-refinement-dev` PR) and disturbing the calculation hp refinement candidates.

Reply:
It seems the relaxed case is useful for studying convergence and understanding the hp algorithm, which justifies having this feature, but in practice I don't see how it would be useful for solving PDEs, in which case one would want full discontinuity (L2) or full continuity (H1), not something in between.

Reply:
Maybe it could be a kind of non-conforming method for H1 like Crouzeix-Raviart?

Reply:
I think both options can be useful in practice. The users have the option to try both and decide for themselves.